### DISCUSSION TITLE
Can include_in_schema support the default false, endpoints and routers settings true?

### DISCUSSION THREAD

**Discussion by stilleshan** [fastapi/fastapi#9226] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
# main.py
app = FastAPI(include_in_schema=False)
app.include_router(api_router)

# api/__init__.py
api_router = APIRouter()
api_router.include_router(public_router, include_in_schema=True)
api_router.include_router(private1_router)
api_router.include_router(private2_router)
```

### Description

I hope to set the global default to not display the endpoints or routers information in the document like the above Example Code, but to set the public in the routers or endpoints separately.

I searched the Internet and official documents, and even asked ChatGPT, and the answer was no, it can only be displayed by default and hidden separately.

Will this requirement be supported in a future release?

Thanks.

### Operating System

macOS

### FastAPI Version

0.92.0

### Python Version

3.9

**Comment by sanchezfdezjavier** [comment]
Same here!

**Comment by aviperl-100boulevard** [comment]
+1 for this idea

**Comment by cbetta** [comment]
Would love to see this. Was this ever added?
